react/components/FilterSidebar.js

```diff
@@ -1,159 +1,152 @@
 import PropTypes from 'prop-types'
-import {
-  map as mapRamda,
-  flatten,
-  filter,
-  pipe,
-} from 'ramda'
-import React, { Component, Fragment } from 'react'
+import { map, flatten, filter, pipe } from 'ramda'
+import React, { useRef, useState, useEffect, Fragment } from 'react'
 import { FormattedMessage } from 'react-intl'
 
-import { withRuntimeContext, Link } from 'vtex.render-runtime'
+import { useRuntime } from 'vtex.render-runtime'
 import { Button } from 'vtex.styleguide'
 import { IconFilter } from 'vtex.dreamstore-icons'
 
-import { facetOptionShape } from '../constants/propTypes'
-import { mountOptions } from '../constants/SearchHelpers'
 import AccordionFilterContainer from './AccordionFilterContainer'
 import Sidebar from './SideBar'
+import { facetOptionShape } from '../constants/propTypes'
+import { getMapByType } from '../constants/SearchHelpers'
 
 import searchResult from '../searchResult.css'
 
-class FilterSidebar extends Component {
-  static propTypes = {
-    map: PropTypes.string.isRequired,
-    rest: PropTypes.string.isRequired,
-    getLinkProps: PropTypes.func.isRequired,
-    runtime: PropTypes.object.isRequired,
-    filters: PropTypes.arrayOf(facetOptionShape).isRequired,
-    selectedFilters: PropTypes.arrayOf(facetOptionShape).isRequired,
-  }
+const FilterSidebar = ({ filters }) => {
+  const { navigate } = useRuntime()
 
-  state = {
-    openContent: false,
-    selectedFilters: pipe(
-      mapRamda(filter =>
-        mountOptions(filter.options, filter.type, this.props.map, this.props.rest),
-      ),
+  const [open, setOpen] = useState(false)
+  const [selectedFilters, setSelectedFilters] = useState(() =>
+    pipe(
+      map(filter => filter.facets),
       flatten,
-      filter(option => option.selected),
-      mapRamda(option => ({
-        ...option,
-        link: option.Link,
-      }))
-    )(this.props.filters || []),
-  }
+      filter(facet => facet.selected)
+    )(filters || [])
+  )
+
+  const lastSelectedFilters = useRef(selectedFilters)
 
-  handleFilterCheck = filter => {
-    if (!this.isOptionSelected(filter)) {
-      this.setState({
-        selectedFilters: this.state.selectedFilters.concat({ ...filter, link: filter.Link }),
-      })
+  useEffect(() => {
+    if (!open) {
+      lastSelectedFilters.current = selectedFilters
+    }
+  }, [selectedFilters, open])
+
+  const isOptionSelected = opt =>
+    !!selectedFilters.find(facet => facet.slug === opt.slug)
+
+  const handleFilterCheck = filter => {
+    if (!isOptionSelected(filter)) {
+      setSelectedFilters(selectedFilters.concat(filter))
     } else {
-      this.setState({
-        selectedFilters: this.state
-          .selectedFilters
-          .filter(opt => opt.slug !== filter.slug),
-      })
+      setSelectedFilters(
+        selectedFilters.filter(facet => facet.slug !== filter.slug)
+      )
     }
   }
 
-  handleClose = () => {
+  const handleClose = () => {
     document.body.style.overflow = 'visible'
```

Comment:
Necessary? Take a minute to think about it...

Reply:
yeah, it is